Userland/Libraries/LibWeb/Fetch/Infrastructure/PortBlocking.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2022, Linus Groh <linusg@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Array.h>
+#include <LibWeb/Fetch/Infrastructure/HTTP/Requests.h>
+#include <LibWeb/Fetch/Infrastructure/PortBlocking.h>
+#include <LibWeb/Fetch/Infrastructure/URL.h>
+
+namespace Web::Fetch::Infrastructure {
+
+// https://fetch.spec.whatwg.org/#block-bad-port
+RequestOrResponseBlocking block_bad_port(NonnullRefPtr<Request> request)
```

Comment:
Done, thanks for pointing these out! Fixed up:

- `Request::appropriate_network_error()`
- `should_response_to_request_be_blocked_due_to_its_mime_type()`
- `should_response_to_request_be_blocked_due_to_nosniff()`
- `block_bad_port()`
- `determine_requests_referrer()`

to take const refs instead of NNRPs.